Add support for UPS's TimeInTransit API

eboswort opened the pull request:
This PR is one of the options I brought up in #242 to add support for UPS's time in transit API.
 
It adds support for accurate time in transit estimates for UPS shipments.

wvanbergen commented on lib/active_shipping/carriers/ups.rb:
```diff
@@ -164,6 +169,16 @@ def create_shipment(origin, destination, packages, options = {})
       end
     end
 
+    def get_delivery_dates(origin, destination, packages, pickup_date=Date.current, options = {})
```
Can you move pickup date into `options`? This way the method call is compatible with `find_rates`, which would allow us to merge them at some point. Given that it's optional it makes more sense to have in the options hash anyway.

eboswort replied:
I'm thinking this no longer applies given our discussion on #242

wvanbergen commented on lib/active_shipping/delivery_date_estimate.rb:
```diff
@@ -0,0 +1,19 @@
+module ActiveShipping
+  class DeliveryDateEstimate
```
Does this API call also include rates?

I'd prefer it if this class inherits from `RateResponse`, or at least make it duck type compatible with RateResponse for the same reason I noted above.

garethson replied:
AFAIK It won't include pricing info so I'm ok with this new class.

eboswort replied:
Yea it does not have pricing info.

garethson commented on lib/active_shipping/carriers/ups.rb:
```diff
@@ -684,6 +735,28 @@ def parse_tracking_response(response, options = {})
+
+      if success
+        xml.css('ServiceSummary').each do |service_summary|
+          # vvvvv Translate the Time in Transit Codes to the service codes used elsewhere vvvvv
```
`vvvvv` probably a typo?

eboswort replied:
haha I actually just put that in to draw attention to that comment since I was pissed off that UPS decided to use two different sets of service codes in the API. But I'll take that out.

eboswort commented:
Ok I renamed get_delivery_dates to get_delivery_date_estimates since that seemed a bit more clear. Also I removed the "vvvvvv"'s from that comment.

Any other comments?

wvanbergen commented:
LGTM.
